lib/utils/getSelectorMeaningfulSource.mjs

```diff
@@ -0,0 +1,45 @@
+import selectorParser from 'postcss-selector-parser';
+
+/**
+ * Get the source index of a selector, excluding leading whitespace.
+ *
+ * @param {import('postcss-selector-parser').Selector} node
+ *
+ * @returns {{selectorStr: string, index: number, endIndex: number}}
+ */
+export default function getSelectorMeaningfulSource(node) {
+	const firstIndex = node.nodes.findIndex((x) => !selectorParser.isComment(x));
+	const lastIndex = node.nodes.findLastIndex((x) => !selectorParser.isComment(x));
+
+	const first = node.nodes[firstIndex];
+	let last = node.nodes[lastIndex];
+
+	if (!first || !last) {
+		const selectorStr = node.toString().trim();
+		const index = node.sourceIndex ?? 0;
+		const endIndex = index + selectorStr.length;
+
+		return {
+			selectorStr,
```

Comment:
- `selector`
- `selectorString`
- `source`
- `representation`

?

Reply:
No need to go hungarian notation; we have JSDoc.
`selector` seems fine to me.

Reply:
No need to go hungarian notation; we have JSDoc.
Maybe `value`?
If anything it's probably the shape of the object returned that we could bikeshed.

Reply:
`selectorStr` is often used by the caller:

```js
const { selectorStr, index, endIndex } = getSelectorMeaningfulSource(selectorNode);

report({
	ruleName,
	result,
	node: ruleNode,
	message: messages.expected,
	messageArgs: [selectorStr, primary],
	index,
	endIndex,
});
```

Because we use destructuring so much it is indeed important to pick a good name here that doesn't isn't already in use for other variables in rules.

Reply:
I prefer a bit `selector` over `selectorStr` or `value`. And we can use `selectorStr` in destructuring like this:

```js
const { selector: selectorStr, index, endIndex } = getSelectorMeaningfulSource(selectorNode);
```

But this is not a strong opinion. 👍🏼

Reply:
`selector` is fine too.

Reply:
`selector` it is!